script/scrape_lc_discuss_company.py

```
@@ -17,56 +26,201 @@
+# blocked rather than throttled; blind and hn are relaxed but not free.
+DELAYS = {"leetcode": 2.5, "reddit": 8.0, "blind": 2.0, "hn": 1.0}
+
+
+# --------------------------------------------------------------------------- net
+def fetch(url, delay, data=None, headers=None, tries=4):
+    """HTTP GET (or POST when `data` is given), with backoff. Returns bytes, or None
+    when the request is hopeless (4xx that a retry cannot fix, or tries exhausted)."""
+    hdrs = dict(HEADERS, **(headers or {}))
+    for attempt in range(tries):
+        try:
+            req = urllib.request.Request(url, data=data, headers=hdrs)
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)